src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -1032,9 +1096,18 @@ PosList ExpressionEvaluator::evaluate_expression_to_pos_list(const AbstractExpre
           });
         } break;
 
-        case PredicateCondition::Between:
+        case PredicateCondition::BetweenInclusive:
           return evaluate_expression_to_pos_list(*rewrite_between_expression(expression));
 
+        case PredicateCondition::BetweenLowerExclusive:
```

Comment:
once you just have a single rewrite method there is less code here as well

Reply:
should be fixed